test/unittest/tests/packing/test_intpack.cpp

```diff
@@ -0,0 +1,416 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * This file unit tests the macros and functions contained in intpack_inline.h
+ */
+
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+
+/*
+ * wt_size_check_pack_wrapper --
+ *     The WT_SIZE_CHECK_PACK() macro which will directly call return on failure. Creating a wrapper
+ *     function thereby ensures that the macro's return call is restricted to this function's scope.
+ */
+static int
+wt_size_check_pack_wrapper(int value, size_t maxValue)
+{
+    WT_SIZE_CHECK_PACK(value, maxValue);
+    return 0;
+}
+
+/*
+ * wt_size_check_unpack_wrapper --
+ *     The WT_SIZE_CHECK_UNPACK() macro which will directly call return on failure. Creating a
+ *     wrapper function thereby ensures that the macro's return call is restricted to this
+ *     function's scope.
+ */
+static int
+wt_size_check_unpack_wrapper(int value, size_t maxValue)
+{
+    WT_SIZE_CHECK_UNPACK(value, maxValue);
+    return 0;
+}
+
+/*
+ * wt_leading_zeros_wrapper --
+ *     This function wraps WT_LEADING_ZEROS() to create a function that returns the number of
+ *     leading zeros, rather than requiring a result variable to be passed in
+ */
+template <class T>
+static int
+wt_leading_zeros_wrapper(T value)
+{
+    int result = 0;
+    WT_LEADING_ZEROS(value, result);
+    return result;
+}
+
+static void
+unpack_posint_and_check(std::vector<uint8_t> const &packed, uint64_t expectedValue)
```

Comment:
The `const` is in front of the `&`, is that intended? I usually see `const std::vector<uint8_t> &packed` since we are not modifying the vector itself. The question is applicable to the other functions below.

Reply:
I think this is part of a broader question of defining our C++ style guide. I think @jeremythorp was looking into this?